src/popup.jsx

```diff
@@ -54,20 +59,28 @@ function Popup() {
       ) : (
         <>
           <p>Your public key:</p>
-          <div className="input public-key">
-            <code>{key}</code>
-            <button className="button-onlyicon" onClick={clipboardCopyPubKey}>
-              <CopyIcon />
-            </button>
-          </div>
-          <div className="input public-key">
-            <code>{keyNIP19}</code>
-            <button
-              className="button-onlyicon"
-              onClick={clipboardCopyPubKeyNIP19}
-            >
-              <CopyIcon />
-            </button>
+          <div className="public-key">
+            {selectedKey === 'hex' ? <div className="input">
+              <code>{`${key.substring(0, 15)}…${key.substring((key.length - 10))}`}</code>
+              <button className="button-onlyicon" onClick={clipboardCopyPubKey}>
+                <CopyIcon />
+              </button>
+            </div> :
+            <div className="input">
+              <code>{`${keyNIP19.substring(0, 15)}…${keyNIP19.substring((keyNIP19.length -10))}`}</code>
+              <button
+                className="button-onlyicon"
+                onClick={clipboardCopyPubKeyNIP19}
+              >
+                <CopyIcon />
+              </button>
+            </div> }
+            <div className="select key-options">
+              <select value={selectedKey} onChange={handleChange}>
+                <option value="hex">hex</option>
+                <option value="npub">npub</option>
+              </select>
+            </div>
```

Comment:
I'm thinking that when the key type changes, it's only a portion of this that changes. To be more specific, only the value inside.
Could you refactor this to reduce the lines of code and just focus the condition on the actual piece that changes?